File: TapperGeoChallenge/app/src/main/java/com/mapbox/tappergeochallenge/activity/MainMenuActivity.java
package com.mapbox.tappergeochallenge.activity;

import android.content.DialogInterface;
import android.content.Intent;
import android.os.Bundle;
import android.support.v7.app.AlertDialog;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.widget.EditText;

import com.mapbox.tappergeochallenge.R;

import butterknife.ButterKnife;
import butterknife.OnClick;

import static com.mapbox.tappergeochallenge.StringConstants.ONE_PLAYER_GAME;
import static com.mapbox.tappergeochallenge.StringConstants.PLAYER_ONE_NAME;
import static com.mapbox.tappergeochallenge.StringConstants.PLAYER_TWO_NAME;
import static com.mapbox.tappergeochallenge.StringConstants.TWO_PLAYER_GAME;
import static com.mapbox.tappergeochallenge.StringConstants.TYPE_OF_GAME;

public class MainMenuActivity extends AppCompatActivity {

  private String TAG = "MainActivity";

  @Override
  protected void onCreate(Bundle savedInstanceState) {
    super.onCreate(savedInstanceState);
    setContentView(R.layout.activity_main_menu);
    ButterKnife.bind(this);
  }

  @OnClick(R.id.single_player_game_button)
  public void singlePlayerGame(View view) {
    Log.d(TAG, "singlePlayerGame: ");
    Intent intent = new Intent(getApplicationContext(), GameActivity.class);
    intent.putExtra(TYPE_OF_GAME, ONE_PLAYER_GAME);
    startActivity(intent);
  }

  @OnClick(R.id.double_player_game_button)
  public void twoPlayerGame(View view) {
    askAndSetPlayerNames();
  }

  private void askAndSetPlayerNames() {
    AlertDialog.Builder dialogBuilder = new AlertDialog.Builder(this);
    LayoutInflater inflater = this.getLayoutInflater();
    final View dialogView = inflater.inflate(R.layout.player_name_dialog, null);
    dialogBuilder.setView(dialogView);
    final EditText playerOneName = dialogView.findViewById(R.id.player_one_editText_name);
    final EditText playerTwoName = dialogView.findViewById(R.id.player_two_editText_name);
    dialogBuilder.setTitle(R.string.dialog_title);
    dialogBuilder.setPositiveButton(getString(R.string.dialog_positive_button), new DialogInterface.OnClickListener() {
      public void onClick(DialogInterface dialog, int whichButton) {
        Log.d(TAG, "twoPlayerGame: ");
        Intent intent = new Intent(getApplicationContext(), GameActivity.class);
        intent.putExtra(TYPE_OF_GAME, TWO_PLAYER_GAME);
        intent.putExtra(PLAYER_ONE_NAME, playerOneName.getText().toString());
        intent.putExtra(PLAYER_TWO_NAME, playerTwoName.getText().toString());
        startActivity(intent);
      }
    });
    AlertDialog nameDialog = dialogBuilder.create();
    nameDialog.show();
  }
}

File: TapperGeoChallenge/app/src/main/java/com/mapbox/tappergeochallenge/activity/MainMenuActivity.kt
package com.mapbox.tappergeochallenge.activity

import android.content.Intent
import android.os.Bundle
import android.widget.EditText
import androidx.appcompat.app.AlertDialog
import androidx.appcompat.app.AppCompatActivity
import com.mapbox.tappergeochallenge.R
import com.mapbox.tappergeochallenge.StringConstants
import kotlinx.android.synthetic.main.activity_main_menu.single_player_game_button
import kotlinx.android.synthetic.main.activity_main_menu.two_player_game_button

/**
 * This class handles the selection of a one or two player game,
 * entering the two players' names, and any other pre-game
 * initialization that might be needed.
 */
class MainMenuActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main_menu)
        single_player_game_button.setOnClickListener {
            val intent = Intent(applicationContext, GameActivity::class.java)
            intent.putExtra(StringConstants.TYPE_OF_GAME_KEY, StringConstants.ONE_PLAYER_GAME)
            startActivity(intent)
        }
        two_player_game_button.setOnClickListener {
            showPlayerNameDialog()
        }
    }

    private fun showPlayerNameDialog() {
        val dialogView = this.layoutInflater.inflate(R.layout.player_name_dialog, null)
        AlertDialog.Builder(this).apply {
            setView(dialogView)
            val playerOneName = dialogView.findViewById<EditText>(R.id.player_one_editText_name)
            val playerTwoName = dialogView.findViewById<EditText>(R.id.player_two_editText_name)
            setTitle(R.string.dialog_title)
            setPositiveButton(getString(R.string.dialog_positive_button)) { dialog, whichButton ->
                Intent(applicationContext, GameActivity::class.java).apply {
                    putExtra(StringConstants.TYPE_OF_GAME_KEY, StringConstants.TWO_PLAYER_GAME)
                    putExtra(StringConstants.PLAYER_ONE_NAME_KEY, playerOneName.text.toString())
                    putExtra(StringConstants.PLAYER_TWO_NAME_KEY, playerTwoName.text.toString())
                    startActivity(this)
                }
            }
            create().show()
        }
    }
}